src/hdmf/common/resources.py

```diff
@@ -408,7 +409,29 @@ def _get_file_from_container(self, **kwargs):
                 msg = 'Could not find file. Add container to the file.'
                 raise ValueError(msg)
 
-    @docval({'name': 'root_container',  'type': HERDManager,
+    @docval({'name': 'objects', 'type': list,
+             'doc': 'List of objects to check for TermSetWrapper within the fields.'})
+    def __check_termset_wrapper(self, **kwargs):
+        """
+        Takes a list of objects and checks the fields for TermSetWrapper.
+        :return: [[object1, attribute_name1, wrapper1], [object2, attribute_name2, wrapper2], ...]
+        """
+        objects = kwargs['objects']
+
+        ret = [] # list to be returned with the objects, attributes and corresponding termsets
+
+        for obj in objects:
+            # Get all the fields, parse out the methods and internal variables
+            obj_fields = [a for a in dir(obj) if not a.startswith('_') and not callable(getattr(obj, a))]
+
+            for attribute in obj_fields:
+                attr = getattr(obj, attribute)
+                if isinstance(attr, TermSetWrapper):
+                    # Search objects that are wrapped
+                    ret.append([obj, attribute, attr])
```

Comment:
Instead of a list of lists, a slightly more Pythonic way would be to use a list of `namedtuple` https://docs.python.org/3/library/collections.html#collections.namedtuple The main advantage being is that is enforces the type of objects the list contains and it documents what each item in the list and item within the tuple means.

Reply:
Done